bson/bsoncodec/default_value_encoders_test.go

```diff
@@ -237,11 +249,11 @@ func TestDefaultValueEncoders(t *testing.T) {
 				},
 				{
 					"Lookup Error",
-					map[string]interface{}{},
+					map[string]interface{}{"foo": nil},
 					&EncodeContext{Registry: NewRegistryBuilder().Build()},
 					&bsonrwtest.ValueReaderWriter{},
 					bsonrwtest.WriteDocument,
-					ErrNoEncoder{Type: reflect.TypeOf((*interface{})(nil)).Elem()},
+					fmt.Errorf("cannot encode invalid element"),
```

Comment:
I might just be misunderstanding this, but why is this an error? The input seems to reasonably be able to marshal to `{"foo": nil}`.

Reply:
We can't encode nil.  While we can discuss changing that, it would be outside the scope of this ticket.